numpy/core/src/multiarray/ctors.c

```diff
@@ -713,14 +713,20 @@ discover_itemsize(PyObject *s, int nd, int *itemsize, int string_type)
     return 0;
 }
 
+enum NDISOVER_RET {
```

Comment:
```suggestion
enum NDISCOVER_RET {
```

Any reason for the `N` prefix? Perhaps `NPY_` would be better to match `NPY_CASTING`.

Note also that elsewhere we use `typedef enum { ... } name;` instead of this construct.

Reply:
adopted `discovered_t`